extensions/common/metadata_object.cc

```diff
@@ -101,6 +101,11 @@ WorkloadMetadataObject::serializeAsPairs() const {
   if (!app_version_.empty()) {
     parts.push_back({AppVersionToken, app_version_});
   }
+  if (!labels_.empty()) {
+    for (const auto& l : labels_) {
+      parts.push_back({absl::string_view(l.first), absl::string_view(l.second)});
```

Comment:
this makes it impossible to distinguish between a raw label and a well-known field. Since they can make a label with key like `cluster_name_` (for example)

Reply:
yeah, need to bring back `CelStateHashable` instead of `FilterStateObjectWrapper` to support expr like `filter_state.upstream_peer.labels["role"]`.